generate-data/src/main.rs

```diff
@@ -285,74 +281,112 @@ fn build_caniuse_global() -> Result<()> {
-                .into_iter()
-                .map(|(browser, versions)| (decode_browser_name(browser), versions))
-                .collect()
+                let start = versions.len();
+                versions.extend(list.iter().map(|(x, _)| *x));
+                flags.extend(list.iter().map(|(_, y)| *y));
+                let end = versions.len();
+
+                stats.push((browser.as_str(), start, end));
+            }
+            let end = stats.len();
+
+            stats[start..end].sort_by_key(|(browser, ..)| *browser);
+
+            let name = alloc_str(&mut strpool, name);
+            features.push((name, start, end));
         }
 
-        static FEATURES: &[&str] = &[
-            #( #features ),*
-        ];
-        static STATS: [Stat; #features_len] = [
-            #( LazyLock::new(|| stat(include_str!(concat!("features/", #features, ".json")))) ),*
-        ];
+        features.sort_by_key(|(name, ..)| &strpool[(name.0 as usize)..(name.1 as usize)]);
+
+        let (stats_name, stats_list): (Vec<_>, Vec<_>) = stats.iter()
+            .map(|(browser, start, end)| {
+                let browser = encode_browser_name(browser);
+                let start: u32 = (*start).try_into().unwrap();
+                let end: u32 = (*end).try_into().unwrap();
+                (browser, [start, end])
+            })
+            .unzip();
+        let features = features.iter().flat_map(|(name, start, end)| {
+            let name_start = name.0;
+            let name_end = name.1;
+            let start: u32 = (*start).try_into().unwrap();
+            let end: u32 = (*end).try_into().unwrap();
+            quote! {
+                (
+                    PooledStr(#name_start, #name_end),
+                    Feature(#start, #end)
+                )
+            }
+        });
+
+        let version_store_len = write_u32(
+            format!("{OUT_DIR}/caniuse-feature-versionstore.u32seq"),
+            versions.iter().flat_map(|&(start, end)| [start, end])
+        )?;
+        let version_index_len = write_u32(
+            format!("{OUT_DIR}/caniuse-feature-versionindex.u32seq"),
+            stats_list.iter().flatten().copied()
+        )?;
+
+        fs::write(
+            format!("{OUT_DIR}/caniuse-feature-flags.bin"),
+            flags.as_slice()
+        )?;
+        fs::write(
+            format!("{OUT_DIR}/caniuse-feature-browsers.bin"),
+            stats_name.as_slice()
+        )?;
+
+        fs::write(
+            format!("{OUT_DIR}/caniuse-feature-matching.rs"),
+            quote! {
+                static FEATURES: &[(PooledStr, Feature)] = &[#(#features),*];
+
+                // # Safety
+                //
+                // We do the transmute at const context,
+                // and the size and alignment are already checked and guaranteed by compiler.
+                static FEATURES_STAT_VERSION_STORE: &[PairU32; #version_store_len / core::mem::size_of::<PairU32>()] = unsafe {
```

Comment:
Unfortunately, we've introduced unsafe keyword here. but this is the best solution, and it's also easy to confirm that it's safe.

I could list some alternatives, but they all have some flaws
1. Use external libraries like `zerocopy`, `bytemuck` (they do the same thing behind the scenes). this introduces third-party dependencies and causes unnecessary alignment and bounds checks at runtime.
2. Use `struct U32Slice(&'static [u8])` and implement `binary_search` and other algorithms for it. This is complicated and unnecessary.

Reply:
Introducing unsafe code in "generate-data" crate is acceptable, because it's used internally, not published to crates.io.